test/concentrate/parser/gtfs_realtime_enhanced_test.exs

```diff
@@ -178,6 +178,57 @@ defmodule Concentrate.Parser.GTFSRealtimeEnhancedTest do
                  revenue: false
                )
     end
+
+    test "decodes a VehiclePosition JSON map, falling back on trip_id if tm_trip_id is absent" do
+      input = %{
+        "block_id" => "Q238-135",
+        "capacity" => 18,
+        "congestion_level" => nil,
+        "current_status" => "STOPPED_AT",
+        "current_stop_sequence" => 670,
+        "load" => 12,
+        "location_source" => "samsara",
+        "occupancy_percentage" => 0.67,
+        "occupancy_status" => "FEW_SEATS_AVAILABLE",
+        "operator" => %{
+          "id" => build(:operator_id),
+          "logon_time" => 1_534_340_301,
+          "first_name" => build(:first_name),
+          "last_name" => "EVANS"
+        },
+        "position" => %{
+          "bearing" => 135,
+          "latitude" => 42.32951,
+          "longitude" => -71.11109,
+          "odometer" => 5.1,
+          "speed" => 2.9796
+        },
+        "run_id" => "128-1007",
+        "stop_id" => "70257",
+        "timestamp" => 1_534_340_406,
+        "trip" => %{
+          "direction_id" => 0,
+          "overload_offset" => -6,
+          "route_id" => "Green-E",
+          "schedule_relationship" => "SCHEDULED",
+          "start_date" => "20180815",
+          "start_time" => nil,
+          "trip_id" => "37165437-X"
+        },
+        "vehicle" => %{
+          "id" => "G-10098",
+          "label" => "3823-3605",
+          "license_plate" => nil
+        },
+        "revenue" => false
+      }
```

Comment:
Do any factories need to be updated with this? I'm also wondering if this can be condensed via a factory, as it seems like a lot of noise for what fields we're actually testing?

Reply:
We have a factory for `%Realtime.Vehicle{}` but the fields on that struct are staying the same, it's just the data input that's changing and we don't have a factory for that. That said I'll take a quick stab at adding such a factory to make this test more concise.